manifests/server/config.pp

```diff
@@ -22,12 +22,16 @@
   $logbin = pick($options['mysqld']['log-bin'], $options['mysqld']['log_bin'], false)
 
   if $logbin {
-    $logbindir = mysql_dirname($logbin)
-    file { $logbindir:
-      ensure => directory,
-      mode   => '0755',
-      owner  => $options['mysqld']['user'],
-      group  => $options['mysqld']['user'],
+    $logbindir = dirname($logbin)
```

Comment:
This needs to use `mysql_dirname` not `dirname` as the stdlib requirement is > 3.2.0.